pkg/analyzer/analyzers/anthropic/anthropic.go

```diff
@@ -0,0 +1,160 @@
+package anthropic
+
+import (
+	"errors"
+	"os"
+
+	"github.com/fatih/color"
+	"github.com/jedib0t/go-pretty/table"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/analyzers"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/config"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
+)
+
+var _ analyzers.Analyzer = (*Analyzer)(nil)
+
+type Analyzer struct {
+	Cfg *config.Config
+}
+
+// SecretInfo hold the information about the anthropic key
+type SecretInfo struct {
+	Valid              bool
+	Type               string // key type - TODO: Handle Anthropic Admin Keys
+	Reference          string
+	AnthropicResources []AnthropicResource
+	Permissions        string // always full_access
+	Misc               map[string]string
+}
+
+// AnthropicResource is any resource that can be accessed with anthropic key
+type AnthropicResource struct {
+	ID       string
+	Name     string
+	Type     string
+	Metadata map[string]string
+}
+
+func (a Analyzer) Type() analyzers.AnalyzerType {
+	return analyzers.AnalyzerAnthropic
+}
+
+func (a Analyzer) Analyze(_ context.Context, credInfo map[string]string) (*analyzers.AnalyzerResult, error) {
+	key, exist := credInfo["key"]
+	if !exist {
+		return nil, errors.New("key not found in credentials info")
+	}
+
+	secretInfo, err := AnalyzePermissions(a.Cfg, key)
+	if err != nil {
+		return nil, err
+	}
+
+	return secretInfoToAnalyzerResult(secretInfo), nil
+}
+
+func AnalyzeAndPrintPermissions(cfg *config.Config, key string) {
+	info, err := AnalyzePermissions(cfg, key)
+	if err != nil {
+		// just print the error in cli and continue as a partial success
+		color.Red("[x] Error : %s", err.Error())
+	}
+
+	if info == nil {
+		color.Red("[x] Error : %s", "No information found")
+		return
+	}
+
+	if info.Valid {
+		color.Green("[!] Valid Anthropic API key\n\n")
+		// no user information
+		// print full access permission
+		printPermission(info.Permissions)
+		// print resources
+		printAnthropicResources(info.AnthropicResources)
+
+		color.Yellow("\n[i] Expires: Never")
+	}
+}
+
+func AnalyzePermissions(cfg *config.Config, key string) (*SecretInfo, error) {
+	// create a HTTP client
+	client := analyzers.NewAnalyzeClient(cfg)
+
+	var secretInfo = &SecretInfo{}
+
+	secretInfo.Type = "API-Key" // TODO: implement Admin-Key type as well
+
+	if err := listModels(client, key, secretInfo); err != nil {
+		return nil, err
+	}
+
+	if err := listMessageBatches(client, key, secretInfo); err != nil {
+		return nil, err
+	}
+
+	// anthropic key has full access only
+	secretInfo.Permissions = PermissionStrings[FullAccess]
+	secretInfo.Valid = true
+
+	return secretInfo, nil
+}
+
+// secretInfoToAnalyzerResult translate secret info to Analyzer Result
+func secretInfoToAnalyzerResult(info *SecretInfo) *analyzers.AnalyzerResult {
```

Comment:
question: It looks like we collect `AnthropicResources` from `listModels` and `listMessageBatches` in `*SecretInfo`, then iterate over them again to create `Bindings`. Could we skip the intermediary `*SecretItem` and build `Bindings` directly by passing it to `listModels` and `listMessageBatches`? That seems like it would eliminate the extra iteration. I might be missing something, so feel free to correct me if I'm off base.

Reply:
EDIT: After reviewing another analyzer, I don't think we should buck the trend of how these currently work. Please ignore the above comment.